sdk/Sdk.Generators/ExtensionStartupRunnerGenerator.cs

```diff
@@ -0,0 +1,212 @@
+                                                  .Any(c => c.Parameters.Length == 0 &&
+                                                            c.DeclaredAccessibility == Accessibility.Public);
+                if (!constructorExist)
+                {
+                    return $"{typeSymbol.ToDisplayString()} class must have a public parameterless constructor.";
+                }
+
+                // Check the extension startup class implements WorkerExtensionStartup abstract class.
+                if (!(namedTypeSymbol.BaseType!.GetFullName().Equals(startupBaseClassName)))
+                {
+                    return $"{typeSymbol.ToDisplayString()} must be a type which implements {startupBaseClassName}.";
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Writes an assembly attribute with type information about our auto generated WorkerExtensionStartupCodeExecutor class.
+        /// </summary>
+        private static void WriteAssemblyAttribute(IndentedTextWriter textWriter)
+        {
+            textWriter.WriteLine(
+                "[assembly: WorkerExtensionStartupCodeExecutorInfo(typeof(Microsoft.Azure.Functions.Worker.WorkerExtensionStartupCodeExecutor))]");
+        }
+
+        /// <summary>
+        /// Writes a class with code which calls the Configure method on each implementation of participating extensions.
+        /// We also have it implement the same "IWorkerExtensionStartup" interface which extension authors implement.
+        /// </summary>
+        private static void WriteStartupCodeExecutorClass(IndentedTextWriter textWriter, IEnumerable<(string TypeName, string? Error)> types)
+        {
+            textWriter.WriteLine("internal class WorkerExtensionStartupCodeExecutor : WorkerExtensionStartup");
+            textWriter.WriteLine("{");
+            textWriter.Indent++;
+            textWriter.WriteLine("public override void Configure(IFunctionsWorkerApplicationBuilder applicationBuilder)");
+            textWriter.WriteLine("{");
+            textWriter.Indent++;
+
+            foreach (var type in types)
+            {
+                textWriter.WriteLine("try");
+                textWriter.WriteLine("{");
+                textWriter.Indent++;
+                if (type.Error != null)
+                {
+                    textWriter.WriteLine($"throw new InvalidOperationException(\"{type.Error}\");");
+                }
+                else
+                {
+                    textWriter.WriteLine($"new {type.TypeName}().Configure(applicationBuilder);");
+                }
+                textWriter.Indent--;
+                textWriter.WriteLine("}");
+                textWriter.WriteLine("catch (Exception ex)");
+                textWriter.WriteLine("{");
+                textWriter.Indent++;
+                textWriter.WriteLine($"Console.Error.WriteLine(\"Error calling Configure on {type.TypeName} instance.\"+ex.ToString());");
```

Comment:
Since the error message contains "error", the host code will [log this as an error](https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs#L123).